server/server.c

```diff
@@ -13,71 +13,242 @@
+        int new_fd = accept(server_fd, sck_addr, &add_size);
+        if(new_fd < 0)
+        {
+            // perror("accept");
+            syslog(LOG_ERR, "accept: %s", strerror(errno));
+            close(write_to_fd);
+            close(server_fd);
+            freeaddrinfo(new_addr);
+            return EXIT_FAILURE;
+        } 
+        syslog(LOG_INFO, "Accepted connection from %s\n", sck_addr->sa_data);
 
-    printf("New connection fd[%d]\n", new_fd);
-    uint8_t buffer[1024];
-    uint16_t buffer_size = sizeof(buffer)/sizeof(buffer[0]);
-    int ret_val = recv(new_fd, buffer, buffer_size, 0);
-    if(buffer_size == ret_val)
-    {
-        perror("recv");
-    }
-    const char* str = "This is super good stuff\n"; 
-    memcpy(buffer, str, strlen(str));
-    ret_val = send(new_fd, buffer, buffer_size, 0);
-    if(buffer_size == ret_val)
-    {
-        perror("send");
+        printf("New connection fd[%d] %s\n", new_fd, sck_addr->sa_data);
+        uint8_t buffer[1024];
+        uint16_t buffer_size = sizeof(buffer)/sizeof(buffer[0]);
+        memset(buffer, 0, buffer_size);
+
+        uint8_t *buf = (uint8_t*)malloc(1024);
+        buffer_size = 1024;
+        uint16_t total_buf_size = 0, realloc_count = 1;
+        memset(buf, 0, buffer_size);
+        if(buf == NULL)
+        {
+            syslog(LOG_ERR, "ENOMEM");
+            close(write_to_fd);
+            close(server_fd);
+            freeaddrinfo(new_addr);
+            return EXIT_FAILURE;
+        }
+        while (1)
+        {
+            uint8_t found_a_packet = 0;
+            int ret_val = recv(new_fd, &buf[total_buf_size], buffer_size, 0);
+            if (!ret_val)
+            {
+                // No more data to recv
+                break;
+            }
+            else if((buffer_size == ret_val) || (buffer_size != ret_val))
+            {
+
+                printf("\nrecvd: %d bytes\n", ret_val);
+
+                for (size_t i = 0; i < ret_val; i++)
+                {
+                    printf("%c", buf[i]);
+                }
+                printf("\n");
+
+                size_t count = 0, write_size = 0, i = 0;
+                for (i = total_buf_size; i < (total_buf_size+ret_val); i++)
+                {
+                    write_size++;
+              
+                    if(buf[i] == '\n')
+                    {
+                        found_a_packet = 1;
+                        int ret = write(write_to_fd, buf, strlen((char*)buf));
+                        if(ret == strlen((char*)buf))
+                        {
+                            printf("success in writing %ld %ld.\n", count, strlen((char*)buf));
+                        }
+                        break;
+                    }
+                    found_a_packet = 0;
+                }
+
+                if(!found_a_packet)
+                {
+                    realloc_count++;
+                    buf = realloc(buf, realloc_count*buffer_size);
```

Comment:
@sankalp123321 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?